vectordraw/vectordraw.py

```diff
@@ -0,0 +1,546 @@
+        )
+        fragment.add_javascript_url(
+            self.runtime.local_resource_url(self, 'public/js/studio_edit.js')
+        )
+        fragment.add_javascript_url(
+            self.runtime.local_resource_url(self, 'public/js/vectordraw_edit.js')
+        )
+        fragment.initialize_js(
+            'VectorDrawXBlockEdit', {"settings": self.settings}
+        )
+        return fragment
+
+    def validate_field_data(self, validation, data):
+        """
+        Validate this block's field data.
+        """
+        super(VectorDrawXBlock, self).validate_field_data(validation, data)
+
+        def add_error(msg):
+            """ Helper function for adding validation messages. """
+            validation.add(ValidationMessage(ValidationMessage.ERROR, msg))
+
+        if data.background_url.strip():
+            if data.background_width == 0 and data.background_height == 0:
+                add_error(
+                    u"You specified a background image but no width or height. "
+                    "For the image to display, you need to specify a non-zero value "
+                    "for at least one of them."
+                )
+            if not data.background_description.strip():
+                add_error(
+                    u"No background description set. "
+                    "This means that it will be more difficult for non-visual users "
+                    "to solve the problem. "
+                    "Please provide a description that contains sufficient information "
+                    "that would allow anyone to solve the problem if the image did not load."
+                )
+
+    def _validate_check_answer_data(self, data):  # pylint: disable=no-self-use
+        """
+        Validate answer data submitted by user.
+        """
+        # Check vectors
+        vectors = data.get('vectors')
+        if not isinstance(vectors, dict):
+            raise ValueError
+        for vector_data in vectors.values():
+            # Validate vector
+            if not vector_data.viewkeys() == {'tail', 'tip'}:
+                raise ValueError
+            # Validate tip and tail
+            tip = vector_data['tip']
+            tip_valid = isinstance(tip, list) and len(tip) == 2
+            tail = vector_data['tail']
+            tail_valid = isinstance(tail, list) and len(tail) == 2
+            if not (tip_valid and tail_valid):
+                raise ValueError
+        # Check points
+        points = data.get('points')
+        if not isinstance(points, dict):
+            raise ValueError
+        for coords in points.values():
+            # Validate point
+            point_valid = isinstance(coords, list) and len(coords) == 2
+            if not point_valid:
+                raise ValueError
+        # If we get to this point, it means that vector and point data is valid;
+        # the only thing left to check is whether data contains a list of checks:
+        return 'checks' in data
```

Comment:
@singingwolfboy Ah, yes, now that this method raises `ValueError` if it comes across invalid data, it doesn't actually need to return `True` or `False`. So this:

```python
if 'checks' not in data:
    raise ValueError
```
should be enough. Thanks!